include/yaclib/algo/detail/wait_impl.hpp

```diff
@@ -39,6 +41,37 @@ bool Wait(const Timeout& t, Cores&... cs) {
   return ready;
 }
 
+template <typename Timeout, typename... Cores>
+bool Wait(const Timeout& t, Cores&... cs) {
+  static_assert(sizeof...(cs) >= 1, "Number of futures must be at least one");
+  static_assert((... && std::is_same_v<detail::BaseCore, Cores>), "Futures must be Future in Wait function");
+  auto range = [&](auto&& functor) {
+    return (... & functor(cs));  // TODO(MBkkt) Maybe (functor(cs) & ...) is better
```

Comment:
- [ ] Remove TODO

These are the same expressions, the only difference is in associativity, which doesn't matter here and is correct
(... & some) -- (somef1 & somef2) & somef3
(some & ...) -- somef1 & (somef2 & somef3)